src/worker/controller/actions/sync/ReceiveAllPipelineAction.js

```diff
@@ -94,13 +94,28 @@ class ReceiveAllPipelineAction {
         await this._controller.asyncExecCmd(constants.NODE_NOTIFICATIONS.UPDATE_DB, { data: coreMsg });
       }
       this._running = false;
+      this._controller.logger().info(`[SYNC] finished successfully`);
       onEnd(null, allResults);
     } catch (err) {
       this._running = false;
       this._controller.logger().error(`[SYNC] error= ${err}`);
       onEnd(err);
     }
   }
+
+  async asyncExecute(params) {
```

Comment:
Just FYI:
`async` is needed only when `await` is inside the function's body.
An `async` function always returns a `Promise`.
E.g. those three functions are the same:
```javascript
function sum1(a, b) {
  return Promise.resolve(a + b);
}

function sum2(a, b) {
  return new Promise(resolve => {
    resolve(a + b);
  });
}

async function sum3(a, b) {
  return a + b;
}
```

Reply:
Therefore if you know a function returns a `Promise` (like `sum1` and `sum2` above) you can `await` for it.
```javascript
const five = await sum1(2, 3);
const seven = await sum2(1, 6);
```
Also if `sum1` or `sum2` will `reject` the `Promise`, you can catch the reject value from outside when you `await` for the function call.